### ISSUE TITLE
Download C client binary using HTTPS and add checksum verification

### ISSUE THREAD

**Issue by jhecking** [aerospike/aerospike-client-nodejs#153]
The Aerospike addon depends on the Aerospike C client library, which gets downloaded during package installation via HTTP.

    URL="http://artifacts.aerospike.com/aerospike-client-c/${AEROSPIKE_C_VERSION}/${PKG_ARTIFACT}"

In the right conditions (a man in the middle attack for example) an attacker may be able to tamper with these resources. This could mean further compromise of a developer or production system.

**Comment by arch1t3ct** [comment]
I think this broke Mac OS (Sierra) installation:

`error: invalid checksum for 'aerospike-client-c-libuv-devel-4.1.1.pkg'
make: *** [binding_gyp_aerospike_client_c_target_run_scripts_aerospike_client_c_sh.intermediate] Error`

v2.4.1 install with no problems.

**Comment by jhecking** [comment]
I am using macOS Sierra as well but I don't have any problem installing the new package version:

```
$ npm install aerospike

> aerospike@2.4.2 install /Users/jhecking/Projects/node_modules/aerospike
> node-gyp rebuild

  TOUCH binding_gyp_aerospike_client_c_target_run_scripts_aerospike_client_c_sh.intermediate
  ACTION binding_gyp_aerospike_client_c_target_run_scripts_aerospike_client_c_sh binding_gyp_aerospike_client_c_target_run_scripts_aerospike_client_c_sh.intermediate
info: downloading 'https://artifacts.aerospike.com/aerospike-client-c/4.1.1/aerospike-client-c-libuv-devel-4.1.1.pkg' to '/Users/jhecking/Projects/node_modules/aerospike/aerospike-client-c/package/aerospike-client-c-libuv-devel-4.1.1.pkg'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2096k  100 2096k    0     0  11.2M      0 --:--:-- --:--:-- --:--:-- 11.3M
info: verifying checksum for 'aerospike-client-c-libuv-devel-4.1.1.pkg': OK
info: extracting files from 'aerospike-client-c-libuv-devel-4.1.1.pkg'

CHECK
   [✓] /Users/jhecking/Projects/node_modules/aerospike/aerospike-client-c/package/usr/local/lib/libaerospike.a
   [✓] /Users/jhecking/Projects/node_modules/aerospike/aerospike-client-c/package/usr/local/include/aerospike/aerospike.h
   [✓] /Users/jhecking/Projects/node_modules/aerospike/aerospike-client-c/package/usr/local/aerospike/client/sys/udf/lua/aerospike.lua
```

Can you download the C client package manually and run sha256sum on it, please?

```
$ curl -O https://artifacts.aerospike.com/aerospike-client-c/4.1.1/aerospike-client-c-libuv-devel-4.1.1.pkg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2096k  100 2096k    0     0  9999k      0 --:--:-- --:--:-- --:--:--  9.7M
$ sha256sum aerospike-client-c-libuv-devel-4.1.1.pkg
b5daf982c8bb46666287540f89df32f64b80a50d2ab1bc6ff5baad8e143914fb  aerospike-client-c-libuv-devel-4.1.1.pkg
```

**Comment by jhecking** [comment]
Oh, I see what the problem is. The `sha256sum` tool doesn't come with macOS; it's part of the homebrew coreutils package on my system. Looks like I will have to use a different way to verify the checksum on macOS.

Can you let me know if this works on your system:

    $ openssl dgst -sha256 aerospike-client-c-libuv-devel-4.1.1.pkg
    SHA256(aerospike-client-c-libuv-devel-4.1.1.pkg)= b5daf982c8bb46666287540f89df32f64b80a50d2ab1bc6ff5baad8e143914fb

**Comment by arch1t3ct** [comment]
@jhecking `$openssl dgst -sha256 aerospike-client-c-libuv-devel-4.1.1.pkg` works :)

**Comment by jhecking** [comment]
@arch1t3ct, I have just released v2.4.3 which should now install on macOS without coreutils as well. Please give it a try and let me know if you are still seeing any issues.

**Comment by arch1t3ct** [comment]
It works. Thanks!
